lib/api/core/sdk/funnelProtocol.js

```diff
@@ -22,23 +22,46 @@
 'use strict';
 
 const
-  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter,
-  Request = require('kuzzle-common-objects').Request;
+  {
+    Request,
+    errors: {
+      InternalError: KuzzleInternalError
+    }
+  } = require('kuzzle-common-objects'),
+  User = require('../models/security/user'),
+  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter;
 
 class FunnelProtocol extends KuzzleEventEmitter {
-  constructor(funnel) {
+  constructor(funnel, user = null) {
     super();
+
     this.funnel = funnel;
     this.id = 'funnel';
+    this.user = user;
+
+    if (this.user && !(this.user instanceof User)) {
+      throw new KuzzleInternalError('FunnelProtocol.constructor: Invalid User object');
+    }
   }
 
-  isReady() {
+  isReady () {
     return true;
   }
 
+  /**
+   *  Hydrate the user and execute SDK query
+   */
   query (request) {
-    return this.funnel.executePluginRequest(new Request(request))
-      .then(result => ({result}));
+    const kuzzleRequest = new Request(request, {
+      connection: { protocol: 'funnel' }
+    });
+
+    if (this.user) {
```

Comment:
This condition should be replaced with a proper call to the `Request` object constructor:

```js
const kuzzleRequest = new Request(request, {
  connection: { protocol: 'funnel' },
  user: this.user
});
```